docs/add_data_source.md

```diff
@@ -0,0 +1,151 @@
+# How-To Add a Data Provider
+
+When adding a new data provider and input structure, there are several areas of the system
+that must be updated. The system [settings.yml](../config/settings.yml) file must be updated
+to include your new provider in the harvester jobs. A [configuration file](#config) must be written, and
+optionally a [traject macro](#macro).
+
+## Steps
+
+1. [Add source to](#settings) [settings.yml](../config/settings.yml)
+2. [Establish data mapping](#mapping)
+3. [Write spec to test data mapping and provide fixture data for testing](#test)
+4. [Write mapping configuration](#config)
+5. [[Optional] Write mapping macro for data transformations](#macro)
+
+### Settings <a name="settings" />
+
+1. At the end of the sources section, add your new source.
+2. Give it a meaningful id (see existing examples): `my_new_source`
+3. Assign the directory within the repository that data files will be located (see: [dlme-metadata](https://github.com/waynegraham/dlme-metadata))
+4. If a single file will exists (i.e. for a large CSV file) set the `only` setting. Example: `only: !ruby/regexp '/egyptian-20170820.csv/'`
+5. Assign the traject configuration file: `traject_file: my_new_source_config`
+6. Assign any appropriate properties. Example:
+```
+`agg_provider`: [NAME OF DATA PROVIDER]
+`inst_id`: short id prefix [i.e. `my_source`]
+```
+
+#### Example configuration
+
+```
+stanford_mods:
```

Comment:
I'd suggest grabbing the example with comments that @cmh2166 added'; in particularly, it emphasizes the optional vs required configuration.